Mathlib/RingTheory/Localization/Ideal.lean

```diff
@@ -231,6 +231,32 @@ theorem ideal_eq_iInf_comap_map_away {S : Finset R} (hS : Ideal.span (α := R) S
     rw [pow_add, mul_assoc, ← mul_comm x, e]
     exact I.mul_mem_left _ y.2
 
+variable (R) in
+lemma _root_.NoZeroSMulDivisors_of_isLocalization (Rₚ Sₚ : Type*) [CommRing Rₚ] [CommRing Sₚ]
```

Comment:
Do you mean I don't need addition? `IsLocalization` is defined only for semirings.

Reply:
No, what I meant is that I do not think that you need `S` (and so `S_p`) to be `R`-algebras, they only need to be `R`-modules. I would then expect to use `[IsLocalizedModule](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LocalizedModule/Basic.html#IsLocalizedModule)`  instead of `IsLocalization` for `S_p`; but you would need to manually provide the morphism from `R` to `S`, and perhaps to ask that the image of `M` is a monoid and that it satisfies `NoZeroSmulDivisors`. 

I agree this can look a bit strange, but since the conclusion only pertains to the `R_p`-module structure of `S_p`, it seems to me that you are only using the algebra structures involved to have nice morphisms (which can be achieved by asking that such morphisms exist), and not because you really need `S` or `S_p` to be themselves rings.

Reply:
I get the idea, but in practice I don't see a precise statement. If `S` is not a ring but only an `R`-module, what is the morphism from `R` to `S`? And If `S` is a ring then any localization as `R`-module is automatically a ring.